python-test/features/steps/local_agent.py

```diff
@@ -101,6 +133,27 @@ def check_logs_contain_message(logs, expected_message):
     return False
 
 
+def check_logs_contain_message_for_policy(logs, expected_message, policy):
+    """
+    Checks agent container logs for expected message for applied policy
+
+    :param (list) logs: list of log lines
+    :param (str) expected_message: message that we expect to find in the logs
+    :param (dict) policy: policy
+    :returns: (bool) whether expected message was found in the logs for expected policy
+    """
+
+    for log_line in logs:
+        log_line = safe_load_json(log_line)
+        if log_line is not None and log_line['msg'] == expected_message:
+            if 'id' in log_line.keys() and log_line['id'] == policy['id']:
+                return True
+            if 'policy_id' in log_line.keys() and log_line['policy_id'] == policy['id']:
+                return True
```

Comment:
Why do we have `id` and `policy_id` as possible valid outcomes?

Reply:
And do you think we should add policy-related knowledge to this `local_agent` module?

Reply:
ok the log lines look like this
```
{"level":"info","ts":1640621757.9308138,"caller":"agent/comms.go:142","msg":"completed RPC subscription to group","name":"home","topic":"channels/65e72370-a778-48aa-8d2c-51e13412bad0/messages/fromcore"}
{"level":"info","ts":1640621757.94167,"caller":"policyMgr/manager.go:50","msg":"managing agent policy from core","action":"manage","name":"roku","dataset":"42d8e6dd-ac1a-47e6-b32b-a18746bddb42","backend":"pktvisor","id":"811c5953-9ab3-45be-8353-08136144944a","version":0}
{"level":"info","ts":1640621758.0065007,"caller":"policyMgr/manager.go:146","msg":"policy applied successfully","policy_id":"811c5953-9ab3-45be-8353-08136144944a","policy_name":"roku"}
{"level":"info","ts":1640621758.0065813,"caller":"policyMgr/manager.go:50","msg":"managing agent policy from core","action":"manage","name":"general","dataset":"aec80952-1434-48a0-8d9c-d91c0d9951ce","backend":"pktvisor","id":"0c4818ae-75c8-4359-aaf9-bdd8432cf577","version":0}
{"level":"info","ts":1640621758.0665298,"caller":"policyMgr/manager.go:146","msg":"policy applied successfully","policy_id":"0c4818ae-75c8-4359-aaf9-bdd8432cf577","policy_name":"general"}
```
So I see "managing agent policy from core" logs "id" instead of "policy_id" but this is a bug because we should be staying consistent here. Let's get a ticket open to fix the agent log message to use "policy_id".

However, I think it may also make more sense to look for the "policy applied successfully" message instead of "managing agent policy from core", which does use "policy_id".

Reply:
ok